gwpy/timeseries/timeseries.py

```diff
@@ -1463,18 +1474,31 @@ def taper(self, side='leftright'):
         if side not in ('left', 'right', 'leftright'):
             raise ValueError("side must be one of 'left', 'right', "
                              "or 'leftright'")
+        if duration and nsamples:
+            raise ValueError("only one of 'duration' or 'nsamples' may be "
+                             "provided as arguments")
```

Comment:
I think I would handle all of argument parsing here, something like:

```python
        elif duration:
            nsamples = int(duration * self.sample_rate.to("Hz").value)
```

then you only need to consider `if nsamples` from that point forward.

Reply:
I'll give that a try, maybe it'll help with codeclimate

Reply:
This helped a lot with the cyclomatic complexity, thanks for the suggestion. The cognitive complexity is still showing up as high.